spirv_msl.hpp

```diff
@@ -505,6 +505,13 @@ class CompilerMSL : public CompilerGLSL
 		// Note: Only Apple's GPU compiler takes advantage of the lack of coherency, so make sure to test on Apple GPUs if you disable this.
 		bool readwrite_texture_fences = true;
 
+		// Metal 3.1 introduced a Metal regression bug which causes infinite recursion during 
+		// Metal's analysis of an entry point input structure that is itself recursive. Enabling
+		// this option will replace the recursive input declaration with a alternate variable of
+		// type void*, and then cast to the correct type at the top of the entry point function.
+		// The bug has been reported to Apple, and will hopefully be fixed in future releases.
+		bool replace_recursive_inputs = false;
```

Comment:
Does this really need to be an option? Seems like everyone needs this workaround. Linked list in buffers isn't exactly ... common.

Reply:
I made it an option, because it does not need to be invoked before macOS 14.0 or iOS 17.0 runtimes, _**and**_  I am hoping it should have a short shelf-life, if Apple responds to the regression bug (which by their feedback, I believe they will). So, it may not be needed again soon, and hopefully it will only apply to a short window of OS versions. 

MoltenVK tracks its need, and sets it, based on OS release. 

Invoking it only when it is needed makes sense, as it makes the MSL shaders less readable when it is invoked. And I wasn't sure that other SPIRV-Cross users would need it.

Although we _**might**_ try to internalize this into SPIRV-Cross based on MSL version, it's really tied to the Metal pipeline compiler version itself, not the MSL language version pushed to SPIRV-Cross.

BTW...given it's obscurity, I did contemplate ignoring it and waiting for Apple to fix it, but it does block CTS testing, which significantly hampers MoltenVK development.